python/rmm/rmm/_lib/memory_resource.pyx

```diff
@@ -1188,3 +1188,10 @@ def get_log_filenames():
         else None
         for i, each_mr in _per_device_mrs.items()
     }
+
+
+def get_free_device_memory(percent):
+    """
+    Returns the percentage of the free memory available
```

Comment:
*nit*: I think this docstring is slightly misleading, since we do not return a percentage.

Perhaps:
```
Return the number of bytes of available device memory.

Parameters
----------
percent
    Integer percentage of the available device memory to return.

Returns
-------
Number of available bytes, aligned (down) to the nearest CUDA allocation size.
```

Reply:
I am starting to think that conflating these two things (bytes of free memory, percentage calculation) is leading to confusion. In librmm I provide two functions, but one calls the other implicitly. I think instead it would be clearer to have two functions (in both Python and C++):

```
def percent(percentage: int, value):
  return value * (percentage / 100)

def get_free_device_memory():
  return free_device_memory() #cython call to librmm

bytes = percent(get_free_device_memory(), 80)
```

Or just get rid of the percent function and have everyone do their own division. I added these functions to librmm when we removed the default initial pool size calculations, to make it easy to reproduce previous behavior.

Reply:
The “integer percentage” concept is very awkward to me. I’m in favor of avoiding APIs that express integer percentages, let’s stick to plain bytes.

Reply:
Dropped the percentage parameter, I'll implement the percentage utility in `cudf` https://github.com/rapidsai/cudf/pull/15628, which is where this is necessary.